src/platform-implementation-js/namespaces/app-menu.ts

```diff
@@ -102,11 +104,85 @@ Items with CollapsiblePanels can also have accessories which provide primary act
 */
 export default class AppMenu {
   #driver;
+  #events = stream<
+    {
+      type: 'collapseToggled';
+      /** Whether or not the AppMenuBurger is open or not. */
+      open: boolean;
+      /**
+       * 'start' is when a `transactionstart` event is fired on a collapsible panel.
+       * 'end' is when a `transactioncancel` event is fired on a collapsible panel. */
+      stage: 'start' | 'end';
+    },
+    unknown
+  >((emitter) => {
+    const f = async () => {
+      const isShown = await this.isShown();
+
+      if (!isShown) {
+        return;
+      }
+
+      const appMenu = await GmailElementGetter.getAppMenuAsync();
+
+      if (!appMenu) {
+        return;
+      }
+
+      const onTransition =
+        (stage: 'start' | 'end') =>
+        async ({ target }: TransitionEvent) => {
+          const { TOGGLE_OPEN_STATE } = CollapsiblePanelView.elementCss;
+
+          if (
+            target instanceof HTMLElement &&
+            target.matches(CollapsiblePanelView.elementSelectors.NATIVE) &&
+            ((stage === 'start' &&
+              target.classList.contains(TOGGLE_OPEN_STATE)) ||
+              // TOGGLE_OPEN_STATE is removed on transitioncancel TransitionEvents
+              stage === 'end')
+          ) {
+            emitter.emit({
+              type: 'collapseToggled',
+              open: this.isMenuCollapsed(),
+              stage,
+            });
+          }
+        };
+
+      appMenu.parentElement!.addEventListener(
+        'transitionstart',
+        onTransition('start')
+      );
+      appMenu.parentElement!.addEventListener(
+        'transitioncancel',
+        onTransition('end')
+      );
```

Comment:
I think `transitioncancel` should be handled as well.

Reply:
`transitioncanend` or `transitionend`?

Reply:
At least in the use case we have at Streak, I'd worry we'd end up double firing if something changed within Gmail around how transitions triggered if we handled both `transitioncancel` and `transitionend`.

Reply:
[079ff95](https://github.com/InboxSDK/InboxSDK/pull/936/commits/079ff957cebe68b578391319dd4e2ab8a3d74dc6) handles the three events and adds a caveat that `transitionend` doesn't currently seem to fire in TSDoc.